client/main.js

```diff
@@ -390,18 +390,40 @@ $(document).ready(function() {
   }
 
   $('#nav-wiki-btn').on('click', function(event) {
-      if (!(event.ctrlKey || event.metaKey)) {
-          toggleWiki();
-      }
+    if (!(event.ctrlKey || event.metaKey)) {
+      toggleWiki();
+    }
   });
 
   $('.wiki-aside-action-collapse').on('click', collapseWiki);
 
+  function getWikiPath() {
+    var challengeType = window.common.challengeType;
+    var dashedName = window.common.dashedName;
+    var prefix = '';
+    // Get wiki prefix depending on type of challenge
+    // Types 0, 1 typically have solutions on the wiki prefixed with challenge-
+    // Type 5 typically has a solution on the wiki prefixed with algorithm-
+    if (challengeType === '0' || challengeType === '1') {
+      prefix = 'challenge-';
+      return prefix + dashedName;
+    }
+    if (challengeType === '5') {
+      prefix = 'algorithm-';
+      return prefix + dashedName;
+    }
+    return false;
+  }
+
   function showWiki() {
     if (!main.isWikiAsideLoad) {
       var lang = window.location.toString().match(/\/\w{2}\//);
       lang = (lang) ? lang[0] : '/en/';
       var wikiURL = '//freecodecamp.github.io/wiki' + lang;
+      var wikiPath = getWikiPath();
+      if (wikiPath) {
+        wikiURL = '//freecodecamp.github.io/wiki/en/' + wikiPath;
```

Comment:
So do you have any ideas on this? I've been doing reading and I see a couple potential solutions. There doesn't seem to be simple and reliable client side solution to check whether a page will send a 404 due to cross-origin policy.

1. (Hacky and fragile) Check the iframe's contents upon loading for a 404 message.

2. Send request to FCC's main server with the wiki path to verify that the wiki path is available in that language.

3. Change the wiki server so that it redirects to the english page on a 404.

Let me know if any of these directions sounds good or if I'm missing something obvious. Thanks again for your help.